File: java/L1100/LeetCode1154.java
package L1100;

public class LeetCode1154 {
    private static final int[] dates = new int[]{0, 31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31};
    public int dayOfYear(String date) {
        String[] days = date.split("-");
        int year = Integer.parseInt(days[0]);
        int month = Integer.parseInt(days[1]);
        int day  = Integer.parseInt(days[2]);
        int curDayTh = 0;
        for (int i = 1;i < month;++i) {
            if (i == 2) {
                if ((year % 4 == 0 && year % 100 != 0) || year % 400 == 0) {
                    curDayTh += 29;
                } else {
                    curDayTh += 28;
                }
            } else {
                curDayTh +=  dates[i];
            }
        }
        return curDayTh + day;
    }
}

File: cpp/L1100/LeetCode1154.cc
#include <string>
using std::string;

class Solution {
    bool isLeafYear(int year) {
        return year % 400 == 0 || year % 4 == 0 && year % 100 != 0;
    }
public:
    int dayOfYear(string date) {
        int year{(date[0] - '0') * 1000 + (date[1] - '0') * 100 + (date[2] - '0') * 10 + (date[3] - '0')};
        int month{(date[5] - '0') * 10 + (date[6] - '0')};
        int day{(date[8] - '0') * 10 + (date[9] - '0')};
        int ans{0};
        for (int i = 1;i < month;++i) {
            if (i == 1 || i == 3 || i == 5 || i == 7 || i == 8 || i == 10 || i == 12) {
                ans += 31;
            } else if (i == 4 || i == 6 || i == 9 || i == 11) {
                ans += 30;
            } else if (isLeafYear(year)) {
                ans += 29;
            } else {
                ans += 28;
            }
        }
        return ans + day;
    }
};